cpp/oneapi/dal/algo/dbscan/backend/gpu/kernel_fp_impl.hpp

```diff
@@ -303,104 +672,194 @@ std::int32_t kernels_fp<Float>::start_next_cluster(sycl::queue& queue,
     return start_index.to_host(queue, { index_event }).at(0);
 }
 
-sycl::event set_queue_ptr(sycl::queue& queue,
-                          pr::ndview<std::int32_t, 1>& algo_queue,
-                          pr::ndview<std::int32_t, 1>& queue_front,
-                          std::int32_t start_index,
-                          const bk::event_vector& deps) {
-    ONEDAL_ASSERT(queue_front.get_dimension(0) == 1);
-    auto queue_ptr = algo_queue.get_mutable_data();
-    auto queue_front_ptr = queue_front.get_mutable_data();
+///  A function that sets the init point in ndview<bool, 1>& observation_indices
+///
+/// @param[in]  queue The SYCL queue
+/// @param[in]  observation_indices  The bool array of size `row_count` x `1`
+/// that shows the points(current observations) which should be copied and shared across all ranks
+/// @param[in]  index  The index of the point
+/// @param[in]  value  The value, can be true or false
+/// @param[in]  deps  Events indicating availability of the `data` for reading or writing
+///
+/// @return A SYCL event indicating the availability
+/// of the updated array for reading and writing
+sycl::event set_init_index(sycl::queue& queue,
+                           pr::ndview<bool, 1>& observation_indices,
+                           std::int32_t index,
+                           bool value,
+                           const bk::event_vector& deps) {
+    ONEDAL_PROFILER_TASK(set_init_index, queue);
+
+    auto observation_indices_ptr = observation_indices.get_mutable_data();
 
     return queue.submit([&](sycl::handler& cgh) {
         cgh.depends_on(deps);
-        cgh.parallel_for(bk::make_multiple_nd_range_2d({ 1, 1 }, { 1, 1 }),
-                         [=](sycl::nd_item<2> item) {
-                             queue_ptr[queue_front_ptr[0]] = start_index;
-                             queue_front_ptr[0]++;
-                         });
+        cgh.parallel_for(sycl::range<1>{ std::size_t(1) }, [=](sycl::id<1> idx) {
+            observation_indices_ptr[index] = value;
+        });
     });
 }
 
+///  A function that sets the queue sizes in ndview<std::int32_t, 1>& queue_size
```

Comment:
Need to distinguish somehow the SYCL queue and data point queue.
This description can be misleading.